src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs

```diff
@@ -158,6 +160,34 @@ internal static bool VerifyRestoreOutput(RestoreRequest request, CacheFile cache
                 return false;
             }
 
+            if (request.ForceUpdatePackageLastAccessTime)
+            {
+                foreach (var package in cacheFile.ExpectedPackageFilePaths)
+                {
+                    var packageFolder = new DirectoryInfo(Directory.GetParent(package).FullName).Parent.Parent.FullName;
+
+                    // we avoid fallback folders as they are readonly
+                    if (request.Project.RestoreMetadata.FallbackFolders.Any(fallbackFolder => PathUtility.GetStringComparerBasedOnOS().Equals(fallbackFolder, packageFolder)))
+                        continue;
+
+                    var packageId = new DirectoryInfo(Directory.GetParent(package).FullName).Parent.Name;
+                    var version = Directory.GetParent(package).Name;
```

Comment:
There are multiple invocations of `Directory.GetParent(package)`. Maybe we can call once and reuse the results as needed.

Reply:
Done, thanks.